x-pack/test/reporting_api_integration/reporting_and_security/validation.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import supertest from 'supertest';
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../ftr_provider_context';
+
+// eslint-disable-next-line import/no-default-export
+export default function ({ getService }: FtrProviderContext) {
+  const reportingAPI = getService('reportingAPI');
+  const retry = getService('retry');
+  const log = getService('log');
+  const supertestSvc = getService('supertest');
+
+  const status = (downloadReportPath: string, response: supertest.Response) => {
+    if (response.status === 503) {
+      log.debug(`Report at path ${downloadReportPath} is pending`);
+    } else if (response.status === 200) {
+      log.debug(`Report at path ${downloadReportPath} is complete`);
+    } else {
+      log.debug(`Report at path ${downloadReportPath} returned code ${response.status}`);
+    }
+  };
+
+  describe('Job parameter validation', () => {
+    before(async () => {
+      await reportingAPI.initEcommerce();
+    });
+    after(async () => {
+      await reportingAPI.teardownEcommerce();
+      await reportingAPI.deleteAllReports();
+    });
+
+    describe('printablePdfV2', () => {
+      it('allows width and height to have decimal', async () => {
+        const downloadReportPath = await reportingAPI.postJobJSON(
+          '/api/reporting/generate/printablePdfV2',
+          { jobParams: createPdfV2Params(1541.5999755859375) }
+        );
+
+        await retry.tryForTime(18000, async () => {
+          const response: supertest.Response = await supertestSvc
+            .get(downloadReportPath)
+            .responseType('blob')
+            .set('kbn-xsrf', 'xxx');
+          status(downloadReportPath, response);
+
+          expect(response.status).equal(200);
+        });
+      });
+
+      it('fails if width or height are non-numeric', async () => {
+        const downloadReportPath = await reportingAPI.postJobJSON(
+          '/api/reporting/generate/printablePdfV2',
+          { jobParams: createPdfV2Params('cucucachoo') }
+        );
+        await retry.tryForTime(18000, async () => {
+          const response: supertest.Response = await supertestSvc
+            .get(downloadReportPath)
+            .responseType('blob')
+            .set('kbn-xsrf', 'xxx');
+
+          expect(response.status).equal(500);
```

Comment:
This is a good suggestion, since the test makes it clear that the response code doesn't indicate what the actual problem was. A 400 code would be more semantic, but it would mean that we have to parse and validate the layout portion of the job params from the generation endpoint. Having a pre-check would add more complexity. Alternatively, attempting to generate something and letting it fail isn't optimal and efficient for the user, but it would allow us to keep the code less complex.

The HTTP route for downloading the report file returns a 500 if the report failed. Maybe that should be smarter and look at the actual error code on the job. Reporting knows about its own `ReportingError` types and can map it to a `400` if it makes more sense semantically. That means that the job went further along in execution that it absolutely needed to, but since we don't have a "pre-check" mechanism in Reporting (that could be a big change), it would be easier to make an improvement like this.

Reply:
Gotcha, thanks for explaining that bit. It is true that this PR does not actually update the current behaviour in that respect. Happy to return to this later.

IMO if we want to make a change like this I think it should be done in a way that prevents us from doing work we could have known will fail - so the "pre-check" you were describing.